app/models/registration.rb

```diff
@@ -521,4 +521,56 @@ def competition_events_changed?
   def serializable_hash(options = nil)
     super(DEFAULT_SERIALIZE_OPTIONS.merge(options || {}))
   end
+
+  def attempt_auto_accept
+    return false if Rails.env.production? && EnvConfig.WCA_LIVE_SITE?
+
+    failure_reason = auto_accept_failure_reason
+    if failure_reason.present?
+      log_auto_accept_failure(failure_reason)
+      return false
+    end
+
+    update_payload = build_auto_accept_payload
+    auto_accepted_registration = Registrations::RegistrationChecker.apply_payload(self, update_payload)
+
+    if auto_accepted_registration.valid?
+      update_lanes!(
+        update_payload,
+        'Auto-accept',
+      )
+      true
+    else
+      log_auto_accept_failure(auto_accepted_registration.errors.messages.values.flatten)
+      false
+    end
+  end
+
+  private def auto_accept_failure_reason
+    return 'Competitor still has outstanding registration fees' if outstanding_entry_fees > 0
+    return 'Auto-accept is not enabled for this competition.' unless competition.auto_accept_registrations?
+    return 'Can only auto-accept pending registrations or first position on waiting list' unless competing_status_pending? || (competing_status_waiting_list? && waiting_list_position == 1)
+    return "Competition has reached auto_accept_disable_threshold of #{competition.auto_accept_disable_threshold} registrations" if competition.auto_accept_threshold_reached?
+
+    'Cant auto-accept while registration is not open' unless competition.registration_currently_open?
+  end
+
+  private def log_auto_accept_failure(reason)
+    add_history_entry(
+      { auto_accept_failure_reasons: reason },
+      'System',
+      'Auto accept',
+      'System reject',
```

Comment:
Maybe I18N?

Reply:
Question here: Will this mean that we'll end up saving translated (ie, non-`en`) strings into the registration history? If so, I think we need to be careful about _what_ locale gets stored - a French user signing up for a US comp shouldn't cause the auto-accept message to be in French. 

Something I have been thinking about is saving the translation keys themselves into the registration history, and displaying them localized in the user's browser - but in this case, we should be saving the key, not the i18n'd string

Reply:
Good point! I really don't want to store localized strings in the DB at all, except maybe for human-written `comments` from a Registration or something similar.

Maybe the way to go about this is to pass in values to the reg history that we interpret as "error codes"? That way you'd have a standardized list that you can just "pipe through" in the API and then later localize in the frontend

Reply:
Refactored to push error codes into the reg history - frontend work can happen in a subsequeny PR, that I'd argue isn't even necessary for launch given that their primary use is for us to debug auto accept issues